backend/src/main/resources/db/changelog/db.changelog-master.yaml

```diff
@@ -1715,6 +1715,25 @@ databaseChangeLog:
               migrations_user_id updated_by
             FROM ${database.defaultSchemaName}.person CROSS JOIN migration_user
             WHERE test_result_delivery IS NOT NULL;
+  - changeSet:
+      id: add-test-length
+      author: emma@skylight.digital
+      comment: Add test length column to the device type table. Test length will be based on the model used.
+      validCheckSum: # hopefully this is temporary because I have no idea why the checksum changed. Putting it in so I can test changes
+        - "8:1fc847472c39db08ed0c4428eabbaaa2"
+      changes:
+        - addColumn:
+            tableName: device_type
+            columns:
+              - column:
+                  name: test_length
+                  type: int
+                  remarks: The length (in minutes) that the test will take.
+                  valueNumeric: 15 # backfill the default test length to 15 minutes, since almost all devices have a 15 min test.
+                  defaultValueNumeric: 15
+                  valueComputed: (SELECT name, CASE WHEN name = 'Access io CareStart' THEN 10 END FROM ${database.defaultSchemaName}.device_type)
```

Comment:
Does this actually do the job? It seems like returning two values would confuse somebody...

Reply:
I'm not really sure to be honest! I originally had no `valueNumeric` and my `valueComputed` was 
` (SELECT name, CASE WHEN name = 'Access io CareStart' THEN 10 _ELSE 15_ END FROM ${database.defaultSchemaName}.device_type) `
However, I'm not quite clear if it's possible to "override" the `valueComputed` when creating new rows in the column. The liquibase docs seem to indicate that `value` overwrites all existing rows in the database but doesn't modify the default value, which is what we want, but does the same logic apply to `valueComputed`? If it does, then I'll use the `valueComputed` expression with the `ELSE 15` added back in, remove `valueNumeric`, and keep `defaultValueNumeric`. But if that logic doesn't apply, and new rows added to the database will just rely on this `valueComputed` to determine their value, then we have a bit of an issue.

Reply:
These are the liquibase docs I'm looking at (unfortunately can't hyperlink to specific attributes): https://docs.liquibase.com/change-types/community/add-column.html

Specifically, the note that 

> Setting the value attribute will set all rows existing to the specified value without modifying the column default. Setting the defaultValue attribute will specify a default value for the column.

Reply:
Resolved offline! The solution was to remove `valueNumeric` and change `valueComputed` to only use the `CASE` statement. This should set all existing CareStart devices to have a 10 minute test length, but won't effect the creation of new rows.